### DISCUSSION TITLE
NanoVDB raycasting

### DISCUSSION THREAD

**Discussion by chrisflesher** [NVIDIA/warp#89] [Ideas]
Was wondering if there are any plans to implement raycasting for NanoVDB level sets? Do you think this could have a significant performance improvement over the `mesh_query_ray` function?

**Comment by mmacklin** [comment]
Hi @chrisflesher , it's a good question - if the NVDB is encoding an SDF then I guess using sphere tracing would be a good option. I am not sure this would be significantly faster than the BVH-based ray casting used by `wp.mesh_query_ray()` though.

Let me ask the NVDB team what our options are here.

Best,
Miles

**Reply by chrisflesher** [reply]
Hey Miles, thanks for looking into this, am curious to know if you've heard anything? I was under the impression NanoVDB was supposed to be super fast but not sure how that compares to mesh raytracing.
